installer/lib/daisy/generator.ex

```diff
@@ -0,0 +1,205 @@
+                  when is_list(var!(assigns)) do
+                unquote(compiled)
+              end
+            end
+          else
+            quote do
+              @external_resource unquote(path)
+              def render(unquote(name), unquote(source), _assigns), do: unquote(File.read!(path))
+            end
+          end
+        end
+      end
+
+    quote do
+      unquote(templates_ast)
+      def template_files(name), do: Keyword.fetch!(@templates, name)
+    end
+  end
+
+  defmacro template(name, mappings) do
+    quote do
+      @templates {unquote(name), unquote(mappings)}
+    end
+  end
+
+  def copy_from(%Project{} = project, mod, name) when is_atom(name) do
+    mapping = mod.template_files(name)
+
+    for {format, project_location, files} <- mapping,
+        {source, target_path} <- files,
+        source = to_string(source) do
+      target = Project.join_path(project, project_location, target_path)
+
+      case format do
+        :keep ->
+          File.mkdir_p!(target)
+
+        :text ->
+          create_file(target, mod.render(name, source, project.binding))
+
+        :config ->
+          contents = mod.render(name, source, project.binding)
+          config_inject(Path.dirname(target), Path.basename(target), contents)
+
+        :prod_config ->
+          contents = mod.render(name, source, project.binding)
+          prod_only_config_inject(Path.dirname(target), Path.basename(target), contents)
+
+        :eex ->
+          contents = mod.render(name, source, project.binding)
+          create_file(target, contents)
+      end
+    end
+  end
+
+  def config_inject(path, file, to_inject) do
+    file = Path.join(path, file)
+
+    contents =
+      case File.read(file) do
+        {:ok, bin} -> bin
+        {:error, _} -> "import Config\n"
+      end
+
+    case :binary.split(contents, "import Config") do
+      [left, right] ->
+        write_formatted!(file, [left, to_inject, right])
+
+      [_] ->
+        Mix.raise(~s[Could not find "import Config" in #{inspect(file)}])
+    end
+  end
+
+  def prod_only_config_inject(path, file, to_inject) do
+    file = Path.join(path, file)
+
+    contents =
+      case File.read(file) do
+        {:ok, bin} ->
+          bin
+
+        {:error, _} ->
+          """
+          import Config
+
+          if config_env() == :prod do
+          end
+          """
+      end
+
+    case :binary.split(contents, "if config_env() == :prod do") do
+      [left, right] ->
+        write_formatted!(file, [left, "if config_env() == :prod do\n", to_inject, right])
+
+      [_] ->
+        Mix.raise(~s[Could not find "if config_env() == :prod do" in #{inspect(file)}])
+    end
+  end
+
+  defp write_formatted!(file, contents) do
+    formatted = contents |> IO.iodata_to_binary() |> Code.format_string!()
+    File.mkdir_p!(Path.dirname(file))
+    File.write!(file, [formatted, ?\n])
+  end
+
+  def inject_umbrella_config_defaults(project) do
```

Comment:
I think we are not injecting anything on config, not sure if this function is being used

Reply:
it is not being used yet

Reply:
i think good to have when we implement it in the future

Reply:
[remove unecessary stuff](https://github.com/phcurado/daisy_ui_components/pull/77/commits/4cd5dd9658ff7291b96c19cd1891b74a8ae681df)